pkg/kubectl/cmd/top_pod.go

```diff
@@ -113,9 +124,66 @@ func (o *TopPodOptions) Validate() error {
 
 // RunTop implements all the necessary functionality for top.
 func (o TopPodOptions) RunTopPod() error {
-	metrics, err := o.Client.GetPodMetrics(o.Namespace, o.ResourceName, o.AllNamespaces, o.Selector)
+	var err error
+	selector := labels.Everything()
+	if len(o.Selector) > 0 {
+		selector, err = labels.Parse(o.Selector)
+		if err != nil {
+			return err
+		}
+	}
+	metrics, err := o.Client.GetPodMetrics(o.Namespace, o.ResourceName, o.AllNamespaces, selector)
```

Comment:
Verifying that the resources exist in the API server first lets us avoid the strange error caused by proxy (I wrote about it [here](https://github.com/kubernetes/kubernetes/issues/30818#issuecomment-242434246).
So this error will be checked below - after successful verification. This actually means the case, where there are no pods matching the query and thus no metrics are found - which is a healthy situation.

 This is a little workaround to avoid the message